app/src/main/java/com/jshvarts/todoapp/notelist/NoteListViewModel.kt

```diff
@@ -1,58 +1,62 @@
 package com.jshvarts.todoapp.notelist
 
-import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.viewModelScope
 import com.jshvarts.todoapp.arch.MviViewModel
+import com.jshvarts.todoapp.arch.UiAction
 import com.jshvarts.todoapp.arch.UiResult
 import com.jshvarts.todoapp.arch.asUiResult
+import com.jshvarts.todoapp.data.Note
 import com.jshvarts.todoapp.data.NoteRepository
-import com.jshvarts.todoapp.notelist.ui.NoteListUiAction
-import com.jshvarts.todoapp.notelist.ui.NoteListUiEffect
-import com.jshvarts.todoapp.notelist.ui.NoteListUiState
+import com.jshvarts.todoapp.notelist.ui.*
+import com.jshvarts.todoapp.ui.WhileUiSubscribed
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.*
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-private const val SAVED_STATE_HANDLE_KEY = "NoteListViewModel_uiState_Key"
-
 @HiltViewModel
 class NoteListViewModel @Inject constructor(
-  private val savedStateHandle: SavedStateHandle,
```

Comment:
not using state handle on this screen since the data could get too large to keep in bundle. additionally, the large amount of data can get stale if kept in bundle